pkg/common/cns-lib/volume/manager.go

```diff
@@ -763,3 +765,56 @@ func (m *defaultManager) RelocateVolume(ctx context.Context, relocateSpecList ..
 	}
 	return res, err
 }
+
+// ConfigureVolumeACLs configures net permissions for a given CnsVolumeACLConfigureSpec
+func (m *defaultManager) ConfigureVolumeACLs(ctx context.Context, spec cnstypes.CnsVolumeACLConfigureSpec) error {
+	log := logger.GetLogger(ctx)
+	err := validateManager(ctx, m)
+	if err != nil {
+		return err
+	}
+	// Set up the VC connection
+	err = m.virtualCenter.ConnectCns(ctx)
+	if err != nil {
+		log.Errorf("ConnectCns failed with err: %+v", err)
+		return err
+	}
+
+	var task *object.Task
+	var taskInfo *vim25types.TaskInfo
+
+	task, err = m.virtualCenter.CnsClient.ConfigureVolumeACLs(ctx, spec)
+	if err != nil {
+		log.Errorf("CNS ConfigureVolumeACLs failed from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return err
+	}
+
+	// Get the taskInfo
+	taskInfo, err = cns.GetTaskInfo(ctx, task)
+	if err != nil {
+		log.Errorf("failed to get taskInfo for ConfigureVolumeACLs task from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return err
+	}
+	// Get the taskResult
+	taskResult, err := cns.GetTaskResult(ctx, taskInfo)
+	if err != nil {
+		log.Errorf("unable to find the task result for ConfigureVolumeACLs task from vCenter %q. taskID: %q, opId: %q ConfigureVolumeACL results: %+v",
+			m.virtualCenter.Config.Host, taskInfo.Task.Value, taskInfo.ActivationId, taskResult)
+		return err
+	}
+
+	if taskResult == nil {
+		log.Errorf("taskResult is empty for ConfigureVolumeACLs task: %q", taskInfo.ActivationId)
```

Comment:
Can you log the spec here so that we know for which file volume and spec the API failed?